DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs

```diff
@@ -129,14 +129,16 @@ public DiscordOverwriteBuilder For(DiscordRole role)
         /// </summary>
         /// <param name="other">Overwrite from which data will be used.</param>
         /// <returns>This builder.</returns>
-        public async Task<DiscordOverwriteBuilder> FromAsync(DiscordOverwrite other)
+        public static async Task<DiscordOverwriteBuilder> FromAsync(DiscordOverwrite other)
         {
-            this.Allowed = other.Allowed;
-            this.Denied = other.Denied;
-            this.Type = other.Type;
-            this.Target = this.Type == OverwriteType.Member ? await other.GetMemberAsync().ConfigureAwait(false) as SnowflakeObject : await other.GetRoleAsync().ConfigureAwait(false) as SnowflakeObject;
-
-            return this;
+            return (other.Type switch
+            {
+                OverwriteType.Member => new DiscordOverwriteBuilder(await other.GetMemberAsync().ConfigureAwait(false)),
+                OverwriteType.Role =>  new DiscordOverwriteBuilder(await other.GetRoleAsync().ConfigureAwait(false)),
+                _ => throw new NotImplementedException(),
+            })
+            .Allow(other.Allowed)
+            .Deny(other.Denied);
```

Comment:
This is a breaking change that doesn't add library features, it belongs in 5.x.

Reply:
I have no idea what you are talking about, this change doesn't exist. :joy: